libnmstate/nm/profile.py

```diff
@@ -570,6 +578,9 @@ def classify_profile_for_actions(self, groups):
                     is_absent
                     and self.nmdev.is_software()
                     and self.nmdev.get_device_type() != NM.DeviceType.VETH
+                    # InfiniBand PKEY interface cannot be removed even
+                    # it is software based
+                    and self.iface.type != InterfaceType.INFINIBAND
```

Comment:
Why not? Does it stay forever on the system? What if the user wants to remove it? Is it not possible?

Reply:
I haven't take a deep look on this yet. Just got failure when try to delete it.

Will update it here.

Reply:
Above line has been removed. Somehow pkey can be removed and failure found any more.

Reply:
Thanks!